db/controller.js

```diff
@@ -14,21 +15,21 @@ const getReviews = (request, response, next) => {
   fetchReviews().then((reviews) => {
     response.status(200).send({ reviews });
   })
-    .catch((err) => {
-      console.log(err)
-    });
+    .catch(next)
 };
 
 const getCommentsForReview = (request, response, next) => {
-
   const reviewId = request.params.review_id;
-
-  fetchCommentsForReview(reviewId).then((comments) => {
-    response.status(200).send({ comments });
+  
+Promise.all([fetchCommentsForReview(reviewId), fetchReviewById(reviewId)]).then((comments) => {
+    if (comments[0].length === 0 && comments[1].length === 0) {
+      next();
```

Comment:
further to custom error suggestion, you could Promise.reject and define the status and msg here, instead of invoking a generic next with no argument